kuma/javascript/src/newsletter.jsx

```diff
@@ -89,68 +87,65 @@ export default function Newsletter() {
     const submit = event => {
         let newsletterForm = newsletterFormRef.current;
 
-        // if skipFetch
-        if (newsletterForm) {
-            if (newsletterForm.dataset['skipFetch']) {
-                /*
-                 * An error occured while attempting to subscribe
-                 * the user via Ajax, but no specific error was provided. We
-                 * therefore just send the user directly to the Mozorg
-                 * newsletter subscription page and log the occurence
-                 */
-                ga('send', {
-                    hitType: 'event',
-                    eventCategory: 'newsletter',
-                    eventAction: 'progression',
-                    eventLabel: 'error-forward'
-                });
-                return true;
-            }
-
-            event.preventDefault();
+        if (!newsletterForm) {
+            return;
```

Comment:
Won't this just mean that the `onSubmit` is still firing? I.e. don't you need to `event.preventDefault()` first?

Reply:
This is there for type refinement reasons, meaning the body of that if should never be executed. Unfortunately afaik there's no way to express in our type system that newsletterForm won't ever be null after the first render (which is the first time submit could ever be called).

And if we'd move `preventDefault` up we'd also prevent form submission for the `skipFetch` case where we afaik explicitly want to have it.